src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/ValidateUserTask.cs

```diff
@@ -47,22 +50,28 @@ public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContex
             var outcomes = Roles.Select(x => T[x]).ToList();
             outcomes.Add(T["Authenticated"]);
             outcomes.Add(T["Anonymous"]);
+
             return Outcomes(outcomes.ToArray());
         }
 
         public async override Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
         {
             var user = _httpContextAccessor.HttpContext.User;
             var isAuthenticated = user?.Identity?.IsAuthenticated;
+
             if (isAuthenticated == true)
             {
                 if (SetUser)
                 {
                     workflowContext.Properties["User"] = user;
                 }
-                var outcomes = new List<string>();
-                outcomes.Add("Authenticated");
-                if (Roles.Count > 0)
+
+                var outcomes = new List<string>
```

Comment:
I'm wondering why you add "Authenticated" role here, is it not in user roles?!!

Reply:
Could use the Authenticated role yes. I opted to have it by default and removed the role from the role list.